reconcile/utils/openshift_resource.py

```diff
@@ -431,6 +433,15 @@ def calculate_sha256sum(body):
         return m.hexdigest()
 
 
+def checkExpirationDate(role):
```

Comment:
i'm not sure this file is the right place for this function because it generically handles openshift/kubernetes resouces stuff, while this function is very specific for roles (at least the current usage and the name of the variable indicate that). 

please don't consider this a blocker!

Reply:
Please add a type hint here. Also, as I mentioned above, prefer `snake_case` names instead of `camelCase` for methods, in this case `def check_expiration_date`. Finally, this name doesn't really tell much about the return, why not call it `role_still_valid`? It is totally fine for predicates to be adjectives, and this one tells exactly what to expect as a return value, and calls like above become simpler, like
```python
if not role_still_valid(r):
    raise ValueError(...)
```